libclc/clc/lib/spirv/math/clc_fmin.cl

```diff
@@ -43,8 +48,10 @@ _CLC_DEF _CLC_OVERLOAD half __clc_fmin(half x, half y) {
   return (y < x) ? y : x;
```

Comment:
can we use this same implementation for float and double?

Reply:
It doesn't generate the same IR, because SPIR-V isn't optimized. I wonder if we in fact want to have `half` use `__builtin_fminf16`? We can simplify the definitions if all types are using a builtin. What do you think? That would optimize the `half` case, rather than pessimize the `float`/`double` case.

Reply:
>I wonder if we in fact want to have `half` use `__builtin_fminf16`? We can simplify the definitions if all types are using a builtin. What do you think? That would optimize the `half` case, rather than pessimize the `float`/`double` case.

LGTM. My opinion is that we better use the same implementation for all of the 3 data types. Either using checking nan, or  using builtin function.